docs/madr/decisions/039-transition-to-new-policies.md

```diff
@@ -0,0 +1,644 @@
+  version: 1.0.0
+paths:
+  /meshes/{mesh}/dataplanes/{name}/_snapshot:
+    get:
+      summary: Get a DPP snapshot on a CP
+      description: Returns a DPP snapshot computed by the CP without the actual Envoy call.
+      parameters:
+        - in: path
+          name: mesh
+          required: true
+          description: The mesh of the DPP to get the diff for.
+          schema:
+            type: string
+        - in: path
+          name: name
+          required: true
+          description: The name of the DPP within the mesh to get the diff for.
+          schema:
+            type: string
+        - in: query
+          name: shadow
+          description: |
+            When computing XDS config the CP takes into account policies with 'kuma.io/effect: shadow' label
+          schema:
+            type: boolean
+            default: false
+        - in: query
+          name: include
+          description: |
+            An array of field names that also should be included in the response. By default, only 'xds' field is set by the API server. 
+          schema:
+            type: array
+            items:
+              type: string
+              enum: [diff]
+      responses:
+        '200':
+          description: Successfully retrieved snapshot.
+          content:
+            application/json:
+              schema:
+                type: object
+                required: [xds]
+                properties:
+                  xds:
+                    description: The raw XDS config as an inline JSON object
+                    type: object
+                  diff:
+                    description: |
+                      When '?include=diff' is specified, the field is set to an array of json patches between the current XDS config
+                      and XDS config that takes into account policies with 'kuma.io/effect: shadow'.
+                    type: array
+                    items:
+                      type: object
+                      required:
+                        - op
+                        - path
+                      properties:
+                        op:
+                          type: string
+                          description: Operation to be performed.
+                          enum: [add, remove, replace, move, copy, test]
+                        path:
+                          type: string
+                          description: A JSON Pointer path indicating the part of the document to operate on.
+                        value:
+                          type: object
+                          description: The value to be used within the operations.
+        '400':
+          description: Bad request. Parameters were invalid.
+        '404':
+          description: The specified mesh/name combination was not found.
+```
+
+Endpoint `/_rules` is getting the same `shadow` and `include` query parameters.
```

Comment:
So there's a diff extra field? Where is it because that's the actual response.

https://github.com/kumahq/kuma/blob/9d93621900c57c3569a53924f0ec836c88eba559/api/openapi/specs/api.yaml#L148-L163

Reply:
Discussed [here](https://github.com/kumahq/kuma/pull/9571#discussion_r1542645970) that probably we don't need to re-define `_rules` spec in this MADR. But if it's not clear I can add it

Reply:
Just where do you put the `diff` field?

Reply:
Right, I mention adding it at the root level of `InspectRules` object